pkg/operators/clustermanager/controllers/certrotationcontroller/certrotation_controller.go

```diff
@@ -44,119 +48,220 @@ var ResyncInterval = time.Minute * 5
-	targetRotations      []certrotation.TargetRotation
+	rotationMap          map[string]rotations // key is clusterManager's name, value is a rotations struct
 	kubeClient           kubernetes.Interface
+	secretInformer       corev1informers.SecretInformer
+	configMapInformer    corev1informers.ConfigMapInformer
+	recorder             events.Recorder
 	clusterManagerLister operatorlister.ClusterManagerLister
 }
 
+type rotations struct {
+	signingRotation  certrotation.SigningRotation
+	caBundleRotation certrotation.CABundleRotation
+	targetRotations  []certrotation.TargetRotation
+}
+
 func NewCertRotationController(
 	kubeClient kubernetes.Interface,
 	secretInformer corev1informers.SecretInformer,
 	configMapInformer corev1informers.ConfigMapInformer,
 	clusterManagerInformer operatorinformer.ClusterManagerInformer,
 	recorder events.Recorder,
 ) factory.Controller {
-	signingRotation := certrotation.SigningRotation{
-		Namespace:        helpers.ClusterManagerNamespace,
-		Name:             signerSecret,
-		SignerNamePrefix: signerNamePrefix,
-		Validity:         SigningCertValidity,
-		Lister:           secretInformer.Lister(),
-		Client:           kubeClient.CoreV1(),
-		EventRecorder:    recorder,
-	}
-	caBundleRotation := certrotation.CABundleRotation{
-		Namespace:     helpers.ClusterManagerNamespace,
-		Name:          caBundleConfigmap,
-		Lister:        configMapInformer.Lister(),
-		Client:        kubeClient.CoreV1(),
-		EventRecorder: recorder,
-	}
-	targetRotations := []certrotation.TargetRotation{
-		{
-			Namespace:     helpers.ClusterManagerNamespace,
-			Name:          helpers.RegistrationWebhookSecret,
-			Validity:      TargetCertValidity,
-			HostNames:     []string{fmt.Sprintf("%s.%s.svc", helpers.RegistrationWebhookService, helpers.ClusterManagerNamespace)},
-			Lister:        secretInformer.Lister(),
-			Client:        kubeClient.CoreV1(),
-			EventRecorder: recorder,
-		},
-		{
-			Namespace:     helpers.ClusterManagerNamespace,
-			Name:          helpers.WorkWebhookSecret,
-			Validity:      TargetCertValidity,
-			HostNames:     []string{fmt.Sprintf("%s.%s.svc", helpers.WorkWebhookService, helpers.ClusterManagerNamespace)},
-			Lister:        secretInformer.Lister(),
-			Client:        kubeClient.CoreV1(),
-			EventRecorder: recorder,
-		},
-	}
-
 	c := &certRotationController{
-		signingRotation:      signingRotation,
-		caBundleRotation:     caBundleRotation,
-		targetRotations:      targetRotations,
+		rotationMap:          make(map[string]rotations),
 		kubeClient:           kubeClient,
+		secretInformer:       secretInformer,
+		configMapInformer:    configMapInformer,
+		recorder:             recorder,
 		clusterManagerLister: clusterManagerInformer.Lister(),
 	}
 	return factory.New().
 		ResyncEvery(ResyncInterval).
 		WithSync(c.sync).
-		WithInformers(
-			secretInformer.Informer(),
-			configMapInformer.Informer(),
-			clusterManagerInformer.Informer(),
-		).
+		WithFilteredEventsInformersQueueKeyFunc(
```

Comment:
@elgnay Hi, LeYang, would you have a review of this part? Here we use a specific name to filter the config/secret informer.